spec/unit/application/agent_spec.rb

```diff
@@ -32,563 +32,583 @@
     allow(Puppet::Node.indirection).to receive(:cache_class=)
     allow(Puppet::Node::Facts.indirection).to receive(:terminus_class=)
 
-    expect($stderr).not_to receive(:puts)
-
     allow(Puppet.settings).to receive(:use)
   end
 
-  it "should operate in agent run_mode" do
-    expect(@puppetd.class.run_mode.name).to eq(:agent)
-  end
-
-  it "should declare a main command" do
-    expect(@puppetd).to respond_to(:main)
-  end
-
-  it "should declare a onetime command" do
-    expect(@puppetd).to respond_to(:onetime)
-  end
-
-  it "should declare a fingerprint command" do
-    expect(@puppetd).to respond_to(:fingerprint)
-  end
-
-  it "should declare a preinit block" do
-    expect(@puppetd).to respond_to(:preinit)
-  end
-
-  describe "in preinit" do
-    it "should catch INT" do
-      expect(Signal).to receive(:trap).with(:INT)
-
-      @puppetd.preinit
+  context "with $stderr.puts not expected" do
+    before :each do
+      expect($stderr).not_to receive(:puts)
```

Comment:
The `before` block shouldn't contain an expectation. https://stackoverflow.com/questions/22644648/rails-rspec-expect-method-inside-before-block. Instead I'd add a specific test that verifes stderr is not expected (or just delete the test as it doesn't seem super useful.